### PR TITLE
Add support for making grpc calls to service methods which return streams of responses.

### PR DESCRIPTION
**Pull request by dinowernli**

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -10,6 +10,12 @@ http_file(
   sha256 = "dea9f7e6f540390d60641515a8d078db281bc4cced333dbf229c1525ba842b33",
 )
 
+http_file(
```
Comment (improbable-yifan) [inline WORKSPACE:13]: This is really cool

File: third_party/protobuf/BUILD [hunk 2]
```diff
@@ -7,3 +7,15 @@ java_library(
...
+
+native.genrule(
+    name = "protoc_exe",
+    # TODO(dino): Support other platforms.
+    srcs = [
+        "@protoc_linux_x86_64//file",
+    ],
+    executable = 1,
+    outs = ["protoc"],
+    cmd = "cat $(locations @protoc_linux_x86_64//file) > $@",
```
Comment (improbable-yifan) [inline third_party/protobuf/BUILD:19]: What exactly does this do?

**Comment by ht290** [comment]
Nice. No code change needed?

File: third_party/protobuf/BUILD (hunk 2 above)
Comment (dinowernli) [inline third_party/protobuf/BUILD:19]: This is the recommended workaround for Bazel's current lack of support for marking http_file rules as "executable = 1".

This just takes the file and copies it to a new file which is executable :P

@damienmg

**Comment by dinowernli** [comment]
@ht290 Indeed the code was fine. The problem was that protoc was removed from the stuff shipped with bazel itself, so I needed to add an extra import.

**Pull request merged**